graylog2-web-interface/src/components/lookup-tables/DataAdapterForm.jsx

```diff
@@ -46,21 +53,54 @@ const DataAdapterForm = React.createClass({
     };
   },
 
+  componentWillUnmount() {
+    this._clearTimer();
+  },
+
+  validationCheckTimer: undefined,
+
+  _clearTimer() {
+    if (this.validationCheckTimer !== undefined) {
+      clearTimeout(this.validationCheckTimer);
+      this.validationCheckTimer = undefined;
+    }
+  },
+
+  _validate(adapter) {
+    // first cancel outstanding validation timer, we have new data
+    this._clearTimer();
+    if (this.props.validate) {
+      this.validationCheckTimer = setTimeout(() => this.props.validate(adapter), 500);
+    }
+  },
+
   _onChange(event) {
     const dataAdapter = ObjectUtils.clone(this.state.dataAdapter);
     dataAdapter[event.target.name] = FormsUtils.getValueFromInput(event.target);
-    this.setState({ dataAdapter: dataAdapter });
+    let generateAdapterName = this.state.generateAdapterName;
+    if (generateAdapterName && event.target.name === 'title') {
+      // generate the name
+      dataAdapter.name = this._sanitizeTitle(dataAdapter.title);
```

Comment:
The title is solely for display purposes, the name is for referring to them in pipelines (and other places), correct.

The initial sanitizing is basically just a nicer way of helping the user to fill in that field, I think once the user manually changes it, we should keep it. It also really is just a string and can be anything, the regex is just to make  it  a bit more readable.